optimize performance by avoiding files under .git dirs

villelaitila opened the pull request:
Reduce processing by excluding .git content.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pahen/madge/226)
<!-- Reviewable:end -->

pahen commented:
Sweet :) Would be nice if you could add a test for this too and handle Windows paths too.

villelaitila commented:
> Sweet :) Would be nice if you could add a test for this too and handle Windows paths too.

Adjusted the code to work with Windows paths also, using path.sep now. The same idea might be useful for isNpmPath later, too.

Added also unit tests and adjusted filtering to exclude any matches if filing-cabinet happens to find dependencies that match with files under .git dirs.

pahen commented:
Thanks for your contribution :) Your changes was added in https://github.com/pahen/madge/commit/5f6fdb7624fdad3e79010a1aba50ea0260a51776 and released in Madge 3.5.0
